sdk/src/gateway/client.ts

```diff
@@ -608,84 +610,60 @@ export class GatewayApiClient {
-     * @param {{ signAllInputs: (psbtBase64: string) => Promise<string> }} btcSigner Btc wallet connector
+     * @param {ExecuteQuoteParams} executeQuoteParams - The params to initiate gateway or evm transaction.
      * @param {WalletClient<Transport, ViemChain, Account>} walletClient Wallet client instance
      * @param {PublicClient<Transport>} publicClient Public client instance
+     * @param {{ signAllInputs: (psbtBase64: string) => Promise<string> }} btcSigner - Btc wallet connector
      * @async
      * @returns {Promise<GatewayStartOrder>} The success object.
      */
     async executeQuote(
-        order:
-            | {
-                  type: 'onramp';
-                  quote: GatewayQuote & GatewayTokensInfo;
-                  params: Optional<GatewayQuoteParams, 'toToken' | 'amount'> & {
-                      toUserAddress: Address;
-                      fromUserAddress: string;
-                      fromChain: 'bitcoin';
-                      toChain: 'bob' | 'bob-sepolia';
-                  };
-              }
-            | {
-                  type: 'offramp';
-                  params: Optional<
-                      GatewayQuoteParams,
-                      | 'toChain'
-                      | 'toUserAddress'
-                      | 'affiliateId'
-                      | 'type'
-                      | 'fee'
-                      | 'feeRate'
-                      | 'gasRefill'
-                      | 'strategyAddress'
-                      | 'campaignId'
-                      | 'toToken'
-                      | 'maxSlippage'
-                      | 'fromChain'
-                  > & {
-                      toUserAddress: string;
-                      fromUserAddress: Address;
-                      fromChain: 'bob' | 'bob-sepolia';
-                      toChain: 'bitcoin';
-                  };
-              }
-            | {
-                  type: 'stake';
-                  params: StakeParams;
-              },
-        btcSigner: { signAllInputs: (psbtBase64: string) => Promise<string> },
+        executeQuoteParams: ExecuteQuoteParams,
         walletClient: WalletClient<Transport, ViemChain, Account>,
-        publicClient: PublicClient<Transport>
+        publicClient: PublicClient<Transport>,
+        btcSigner?: { signAllInputs: (psbtBase64: string) => Promise<string> }
     ): Promise<string> {
-        if (order.type === 'onramp') {
-            const { uuid, psbtBase64 } = await this.startOrder(order.quote, order.params);
+        const isOnrampQuoteParams = (args: ExecuteQuoteParams): args is OnRampExecuteQuoteParams => {
+            return args.fromChain.toString().toLowerCase() === 'bitcoin';
+        };
 
-            const bitcoinTxHex = await btcSigner.signAllInputs(psbtBase64!);
+        if (isOnrampQuoteParams(executeQuoteParams)) {
+            const quote = (await this.getQuote(executeQuoteParams)) as GatewayQuote & GatewayTokensInfo;
```

Comment:
Keep the same flow as before, users calls `getQuote` and then `executeQuote` with the result of that